sktime/forecasting/model_selection/tests/test_split.py

```diff
@@ -149,18 +146,25 @@ def _check_cv(cv, y, allow_empty_window=False):
 @pytest.mark.parametrize("window_length", TEST_WINDOW_LENGTHS)
 def test_single_window_splitter(y, fh, window_length):
     """Test SingleWindowSplitter."""
-    cv = SingleWindowSplitter(fh=fh, window_length=window_length)
-    train_windows, test_windows, cutoffs, n_splits = _check_cv(cv, y)
-
-    train_window = train_windows[0]
-    test_window = test_windows[0]
-    assert n_splits == 1
-    assert train_window.shape[0] == _coerce_duration_to_int(
-        duration=window_length, freq="D"
+    fh_horizon = (
+        ForecastingHorizon(fh) if not isinstance(fh, ForecastingHorizon) else fh
```

Comment:
I do not think the test should do this, but the splitter should.
In fact, it is doing this in `get_fh` (but not on construction). Perhaps at some point this should be changed to an attribute, like in `BaseForecaster`.

Reply:
Replaced all instances of
```
    fh_horizon = (
        ForecastingHorizon(fh) if not isinstance(fh, ForecastingHorizon) else fh
    )
```
with
```
check_fh(fh)
```
My bad. Forgot about this small utility function.

Reply:
Still, are we sure we want checks in the tests?
I think the *objects* should be doing this, and not the *tests*.